src/constraint/spec.ts

```diff
@@ -265,9 +274,10 @@ export const SPEC_CONSTRAINTS: SpecConstraintModel[] = [
       if (specM.isAggregate()) {
         let hasNonFacetDim = false, hasDim = false, hasEnumeratedFacetDim = false;
         specM.specQuery.encodings.forEach((encQ, index) => {
-          if (isValueQuery(encQ) || encQ.autoCount === false) return; // skip unused field
+          if (isValueQuery(encQ) || (isAutoCountQuery(encQ) && encQ.autoCount === false)) return; // skip unused field
```

Comment:
We have so many `(isAutoCountQuery(encQ) && encQ.autoCount === false)`. 

I wonder if we should add `isDisabledAutoCountQuery` to make it concise.

Similarly, maybe also `isEnabledAutoCountQuery` for  `(isAutoCountQuery(encQ) && encQ.autoCount === true)`.